subsys/net/lib/nrf_cloud/src/nrf_cloud_fota.c

```diff
@@ -640,8 +717,8 @@ static int start_job(struct nrf_cloud_fota_job * const job)
 	fragment_size = 1024;
 #endif
 
-	ret = fota_download_start(job->host, job->path, sec_tag, NULL,
-				  fragment_size);
+	ret = fota_download_start(job->info.host, job->info.path, sec_tag,
+				  NULL, fragment_size);
```

Comment:
This is what I was looking for when I asked about where the download is done.

Reply:
this is not for BLE FOTA files though.

Reply:
ok, I checked out your repo and branch and now I can see what's to be done.  My BLE callback in the application could call `fota_download_start()` or whatever if the device is connected and ready to receive update fragments.  Something like that anyway.

Reply:
Or submit something to a work queue to actually do it.

Reply:
yes, but it will need to use `download_client`.  `fota_download` is for nrf91 fota (it downloads and applies the update).
a missing note (i will add to the ble fota callback definition), is that the ble job should be copied in full once it is received by the callback.  this module frees the memory once the callback is returned.  the reason for this is that there could be many BLE fota jobs arriving.

Reply:
i'm thinking a `ble_fota_manager` type of module will be needed.  managing fota jobs for many devices is going to require a good bit of logic.  for example, 5 out of 10 ble devices might get a fota job which uses the same file.  or perhaps one ble device only connects intermittently, no need to start that fota when there is a fota for another ble device which is always connected.